### PR TITLE
cleanup sidekick proxy, ui functions

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -534,7 +515,7 @@ func sidekickMain(ctx *cli.Context) {
 		stats := BackendStats{MinLatency: time.Duration(24 * time.Hour), MaxLatency: time.Duration(0)}
 		backend := &Backend{endpoint, proxy, &http.Client{
 			Transport: proxy.Transport,
-		}, false, healthCheckPath, healthCheckDuration, &stats, timeZero}
+		}, 1, healthCheckPath, healthCheckDuration, &stats}
```
Comment (poornas) [inline main.go:518]:
```suggestion
		}, 0, healthCheckPath, healthCheckDuration, &stats}
```

Comment (harshavardhana) [reply main.go]: done

File: main.go [hunk 2]
```diff
@@ -291,66 +290,49 @@ func (b *Backend) updateCallStats(t shortTraceMsg) {
...
-		if proxy != nil {
-			return proxy, idx
+		idx := rng.Intn(len(lb.backends))
+		if lb.backends[idx].IsUp() {
+			// random backend is up.
+			return lb.backends[idx]
 		}
 		tries++
 		if tries == len(lb.backends) {
+			// no backend is up we fail the request.
```
Comment (krishnasrinivas) [inline main.go:307]: previous round robin LB is better. For ex, out of 10 nodes, let's say first 3 are down, then the 4th server gets 40% of the load

Comment (harshavardhana) [reply main.go:307]: Load balancer is not here to optimize down node situation @krishnasrinivas it is meant to evenly spread the requests..

Sidekick's purpose was to do randomized load balancing other than listing calls. Currently sidekick is not doing that, round robin is not optlmal is good situations.

Comment (harshavardhana) [reply main.go:307]: In fact, this was the main requirement of sidekick implementation, if you look at the distribution now you can actually see the best distribution possible even with the downed situation too. 

Due to randomness involved, it allows for a much wider spread. round-robin is generally prone to problems of overloading a single server during disconnects.

Comment (harshavardhana) [reply main.go:307]: I shall add affinity for the number of tries such that tries are not exhausted due to randomness.

Comment (harshavardhana) [reply main.go]: I made more changes @krishnasrinivas PTAL

**Pull request merged**